### PR TITLE
chore(repo): Scoping Turborepo task inputs

### PR DESCRIPTION
**Pull request by tmilewski**
## Description

Speed up DX by scoping down the Turborepo inputs on particular tasks.

As-is, any changed file in a package results in Turborepo marking that package as altered. That slows down the DX as it requires a task (and, potentially, dependent tasks) to be re-run every time.

This PR ensures that only the relevant files are considered when Turborepo is making that determination.

There are a number of useful cases for this, like speeding up the turnaround time for developers running tasks like `npm run test`.

The convoluted example below might not seem like much, but it all adds up, especially, when the dependent tasks are further up the dependency graph.

## Example:

You're testing a change you made to `@clerk/clerk-sdk-node` directly and your editing only your tests.

### Current

You run `npm run test -- --filter=@clerk/clerk-sdk-node`:

Because all files are considered for the dependent `build` task, it's also re-run:

- `@clerk/clerk-sdk-node:build` (runs)
- `@clerk/clerk-sdk-node:test` (runs)

### New

You run `npm run test -- --filter=@clerk/clerk-sdk-node`:

Because only required files are considered for the dependent `build`, it isn't re-run:

- `@clerk/clerk-sdk-node:build` (cached)
- `@clerk/clerk-sdk-node:test` (runs)

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [x] `build/tooling/chore`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,10 +88,10 @@
     "version": "npx changeset version && ./scripts/version-info.sh",
     "version:snapshot": "./scripts/snapshot.mjs",
     "version:staging": "./scripts/staging.mjs",
-    "release": "FORCE_COLOR=1 npm run build -- --force && npx changeset publish && git push --follow-tags",
-    "release:snapshot": "FORCE_COLOR=1 npm run build && npx changeset publish --tag snapshot --no-git-tag",
-    "release:staging": "FORCE_COLOR=1 npm run build && npx changeset publish --tag staging --no-git-tag",
-    "release:verdaccio": "if [ \"$(npm config get registry)\" = \"https://registry.npmjs.org/\" ]; then echo 'Error: Using default registry' && exit 1; else TURBO_CONCURRENCY=4 npm run build && npx changeset publish --no-git-tag; fi",
+    "release": "TURBO_FORCE=true FORCE_COLOR=1 npm run build -- --force && npx changeset publish && git push --follow-tags",
+    "release:snapshot": "TURBO_FORCE=true ORCE_COLOR=1 npm run build && npx changeset publish --tag snapshot --no-git-tag",
```
Comment (LekoArts) [inline package.json:92]:
```suggestion
    "release:snapshot": "TURBO_FORCE=true FORCE_COLOR=1 npm run build && npx changeset publish --tag snapshot --no-git-tag",
```

**Review by LekoArts — CHANGES_REQUESTED** [review]
Small typo

**Review by BRKalow — COMMENTED** [review]
I love that you're using the per-workspace `turbo.json` files here!

File: package.json (hunk 1 above)
Comment (tmilewski) [reply package.json]: Fixed, TY!

File: package.json [hunk 2]
```diff
@@ -1,17 +1,45 @@
...
-  "version": "0.0.0",
-  "private": true,
-  "packageManager": "npm@8.5.0",
-  "engines": {
-    "node": ">=16.8.0",
-    "npm": ">=8.5.0"
+  "scripts": {
+    "build": "FORCE_COLOR=1 turbo build --concurrency=${TURBO_CONCURRENCY:-80%}",
+    "bundlewatch": "turbo bundlewatch",
+    "changeset": "npx changeset",
```
Comment (LekoArts) [inline package.json:14]: If we're changing this file that much anyways, can we remove the `npx` in favor of just calling the scripts like I did here?
https://github.com/clerkinc/javascript/pull/1891#discussion_r1361591281

Then I won't need to deal that much with merge conflicts and the change might actually make more sense in this PR anyways

**Review by BRKalow — COMMENTED** [review]
One question about the `dependsOn` change for `lint`, otherwise this looks pretty good. Nice improvement!

File: turbo.json [hunk 3]
```diff
@@ -56,14 +69,49 @@
...
+        "!coverage/**",
+        "!examples/**",
+        "!node_modules/**"
+      ],
+      "outputMode": "new-only",
       "outputs": []
     },
     "lint": {
-      "dependsOn": ["build"],
+      "dependsOn": ["^build"],
```
[the remark this answers is not in the document]
Comment (tmilewski) [reply turbo.json:113]: @BRKalow Both work; it's just that we technically don't need to build the _current_ package.

**Review by BRKalow — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**